tests/xharness/Execution/ProcessManager.cs

```diff
@@ -59,6 +59,12 @@ public Task<ProcessExecutionResult> RunAsync (Process process, ILog log, TimeSpa
 			return RunAsync (process, log, log, log, timeout, environment_variables, cancellation_token, diagnostics);
 		}
 
+		public Task<ProcessExecutionResult> RunAsync (Process process, MlaunchArguments args, ILog log, TimeSpan? timeout = null, Dictionary<string, string> environment_variables = null, CancellationToken? cancellation_token = null, bool? diagnostics = null)
```

Comment:
This could also set the mlaunch executable path to the process info

Reply:
Yep